src/v/archival/archival_policy.cc

```diff
@@ -241,18 +241,26 @@ static ss::future<> get_file_range(
           "Full segment upload {}, file size: {}",
           upl,
           fsize);
-        co_return;
+        co_return storage::convert_to_file_pos_outcome::success;
     }
     if (begin_inclusive != segment->offsets().base_offset) {
-        auto seek = co_await storage::convert_begin_offset_to_file_pos(
+        auto seek_result = co_await storage::convert_begin_offset_to_file_pos(
           begin_inclusive, segment, upl->base_timestamp, io_priority);
+        if (seek_result.error()) {
+            co_return seek_result.error();
+        }
+        auto seek = seek_result.value();
         upl->starting_offset = seek.offset;
         upl->file_offset = seek.bytes;
         upl->base_timestamp = seek.ts;
     }
     if (end_inclusive) {
-        auto seek = co_await storage::convert_end_offset_to_file_pos(
+        auto seek_result = co_await storage::convert_end_offset_to_file_pos(
           end_inclusive.value(), segment, upl->max_timestamp, io_priority);
+        if (seek_result.error()) {
```

Comment:
Same has_error change as above is needed here

Reply:
fixed - I edited the change into the previous commit